portico: Integrate new /hello page with new navbar and footer.

amanagr opened the pull request:
This was designed and mostly written by Vlad Korobov.

Fixes #24082
<img width="1230" alt="image" src="https://user-images.githubusercontent.com/25124304/216881848-fcaf8efe-f55f-4975-99c4-7d104acaff99.png">

- [x] Link for `Customization` button needs to be decided.
- [x] Delete old css.

I left the navbar in other pages as it is for the reviewer to see how it looks. 

I am not sure if we should integrate this PR with the images of the webapp appearing very different from what it looks like right now, so maybe we need screenshots of how Zulip looks currently and then update it later on? @terpimost FYI

timabbott commented on static/js/portico/landing-page.js:
```diff
@@ -158,11 +208,6 @@ const events = function () {
-
     // Move to the next slide on clicking inside the carousel container
     $(".carousel-inner .item-container").on("click", function (e) {
         const get_tag_name = e.target.tagName.toLowerCase();
```
Presumably this carousel logic can be removed too?

amanagr replied:
Yeah.

timabbott commented on static/styles/portico/footer_new_style.css:
```diff
@@ -0,0 +1,146 @@
+footer {
+    background: #313257;
+    box-sizing: border-box;
+}
```
We can probably call this `footer.css`. We should avoid having CSS for all `<footer>` elements though -- is this meant to be a `.footer`?

amanagr replied:
Ok, used `.footer`.

timabbott commented on static/styles/portico/navbar_new_style.css:
```diff
@@ -0,0 +1,575 @@
+    color: white;
+    z-index: 10;
+    overflow: hidden;
+}
```
This can probably be called just `navbar.css` -- it seems nice to use the cleaner names for the post-redesign version. (I regret having a `new-style` CSS class that's still present in the web app 5 years after being introduced).

amanagr replied:
Done.

timabbott commented:
Thanks for working on this @amanagr!

I posted a few smaller questions, but one thing I'm curious to understand is what the story is with the ~80 image files, often in groups of 4 for a given concept. Is the intent that we'd actually use all of those, or do we just need to link to the `.webp` for each asset? (I could also imagine those just being all the versions a design tool exported).

The CSS is a lot, so I've not tried to review it thoroughly; what's most important is that it looks good and composes sensibly. One thing we definitely want to avoid on the composability front is the extensive use of CSS reset (e.g., having styles for `h2`, rather than for something like `h2.landing-page-heading` -- which means it'll be hard to reason about how this composes with other CSS files that we may want to combine it with (The `footer` CSS file seems designed to compose with other things, but not the other way around?).

amanagr commented:
> I posted a few smaller questions, but one thing I'm curious to understand is what the story is with the ~80 image files, often in groups of 4 for a given concept. Is the intent that we'd actually use all of those, or do we just need to link to the .webp for each asset? (I could also imagine those just being all the versions a design tool exported).

We want to preferably use `webp` images but should have fallback in case the browser doesn't support them. @terpimost FYI I think you should answer this better.

alya commented:
I've managed to get into a state where the menu doesn't have enough room at the moment. I think I can get there pretty reliably by opening the menu on another page, and then pressing the "back" button. The page loads with the menu open, and looking like this:
<img width="1236" alt="Screen Shot 2023-02-05 at 9 37 01 PM" src="https://user-images.githubusercontent.com/2090066/216892294-6dadb154-2f5e-4dad-92f0-ad4d6d090f74.png">

alya commented:
I think clicking outside the menu should close it? Is it intentional that it doesn't?

alya commented:
Playing with the other pages, if I scroll up past the top or down past the bottom, I see a white stripe, which looks a bit odd. I think on the bottom we just want it to be the same color as the nav area. I'm not sure if the top is harder to fix?

alya commented:
I'm seeing some quirks with the bottom nav on other pages (in this case, /features)...

Green hover color:
<img width="202" alt="Screen Shot 2023-02-05 at 10 00 15 PM" src="https://user-images.githubusercontent.com/2090066/216895090-21e33685-00ab-4744-8895-14fa7231711e.png">

Double underline:
<img width="215" alt="Screen Shot 2023-02-05 at 10 00 09 PM" src="https://user-images.githubusercontent.com/2090066/216895093-563d6531-fa56-4746-a48a-23e2a4b39d74.png">

amanagr commented:
> I've managed to get into a state where the menu doesn't have enough room at the moment. I think I can get there pretty reliably by opening the menu on another page, and then pressing the "back" button. The page loads with the menu open, and looking like this:

Fixed.

amanagr commented:
> I think clicking outside the menu should close it? Is it intentional that it doesn't?

Makes sense, added that functionality.

amanagr commented:
> Playing with the other pages, if I scroll up past the top or down past the bottom, I see a white stripe, which looks a bit odd. I think on the bottom we just want it to be the same color as the nav area. I'm not sure if the top is harder to fix? It's not an issue on the /hello page itself.

This only happens on Mac and it is just browser expanding the background color which white for the other pages. I am not sure there is anything I can do here.

amanagr commented:
> I'm seeing some quirks with the bottom nav on other pages (in this case, /features)...


Fixed.

amanagr commented:
@terpimost FYI [this](https://github.com/zulip/zulip/pull/24301#issuecomment-1418515882) needs your feedback.

terpimost commented:
> > Playing with the other pages, if I scroll up past the top or down past the bottom, I see a white stripe, which looks a bit odd. I think on the bottom we just want it to be the same color as the nav area. I'm not sure if the top is harder to fix? It's not an issue on the /hello page itself.
> 
> This only happens on Mac and it is just browser expanding the background color which is white for the other pages. I am not sure there is anything I can do here.

Could you attach screenshot for that case please?


>I think clicking outside the menu should close it? Is it intentional that it doesn't?

It would be nice to discuss it on the mock up stage. I made it in a way because accidental click outside sometimes closes the menu on other websites. But if you feel that click outside the menu should close it like "exit from the modal window" I'm ok with that

terpimost commented:
@amanagr In my version titles are white-ish:
<img width="1395" alt="image" src="https://user-images.githubusercontent.com/1903309/219544788-1420aa2f-ecaa-462e-8741-625010458f6d.png">
In your version they are of a different color:
<img width="1327" alt="image" src="https://user-images.githubusercontent.com/1903309/219544828-2e30ce5f-8518-4cce-a6c5-47ae0e6c1f97.png">


on http://localhost:9991/new/ I see an old header:
<img width="313" alt="image" src="https://user-images.githubusercontent.com/1903309/219544897-6d5602bf-0a9a-459b-8cde-478a80f52143.png">
So I guess this PR is not about header?

It would be nice to do the same trick with other pages so scrolling beyond the bottom of the page wouldn't show different to footer color, like white (Mac nuance):
<img width="1507" alt="image" src="https://user-images.githubusercontent.com/1903309/219545174-bfd3b218-ac33-470b-857f-fc93c30e3ccb.png">
This requires to properly set background color on body and add some background containers..

Copyright is smaller on /why-zulip page 
<img width="387" alt="image" src="https://user-images.githubusercontent.com/1903309/219545355-ddfb0bf6-0c6a-422b-813d-102e4c2e61f1.png">
<img width="322" alt="image" src="https://user-images.githubusercontent.com/1903309/219545375-78f0cb7c-5245-427a-97cd-53199b8233f8.png">

Switching between http://localhost:9991/why-zulip/ and http://localhost:9991/hello/ shows some horizontal shift. Expectation that footers would be exactly the same on the same position.

New organization and Log in links don't work: 
<img width="173" alt="image" src="https://user-images.githubusercontent.com/1903309/219545703-8d777b3e-9e63-4c1a-a6ce-a98ea6ed0f8c.png">


**Case studies** and **Open communities directory** are absolute links - not sure it should be so or not.

Found strange footer here http://localhost:9991/help/getting-started-with-zulip#related-articles 
<img width="836" alt="image" src="https://user-images.githubusercontent.com/1903309/219546139-93b624fa-eb69-4f54-a861-66d5536e67aa.png"> I guess something isn't alright with help center footer.

There is some strange error on the team page:
<img width="1425" alt="image" src="https://user-images.githubusercontent.com/1903309/219546375-2c86382e-9de9-416e-98d9-584bc5c5ead7.png">


About the hello page:
These buttons should open something (where should they link to? @alya ):
<img width="867" alt="image" src="https://user-images.githubusercontent.com/1903309/219546756-14113b02-f2a5-4c9f-a142-f5abe5da6909.png">

All the links to case studies should link to actual page (I'm not sure if that should be absolute link or relative):
<img width="704" alt="image" src="https://user-images.githubusercontent.com/1903309/219546885-cb574e63-97ad-4fb4-abcf-fb9a7b1ef0e8.png">
and in the last block too 
<img width="1245" alt="image" src="https://user-images.githubusercontent.com/1903309/219547076-b3e287d1-077f-4221-b648-d30fb82ca5f7.png">



Links in this block also don't work 
<img width="640" alt="image" src="https://user-images.githubusercontent.com/1903309/219546961-3ea0c159-5b2f-496e-abe2-7145567440ce.png">

I guess it is my mistake I didn't make them actual links, @alya could you please help with that?

amanagr commented:
Thanks, @vlad, for the review, all the footer concerns are already addressed in #24353 which is what we are trying to merge first.

terpimost commented:
Here are modifications I did fo the links of industries:
<img width="1080" alt="image" src="https://user-images.githubusercontent.com/1903309/231292228-efb78fd2-05e4-4495-a8af-cc133f6794b3.png">
- removed space between text and svg icon - space will be added via css
- replaced my `div` tags with `a` - in cc made it display block
- made such css :

```css
.quote__text-industry{
  display: block;
  color: hsl(248 54% 52%);
  font-family: var(--font-ops);
  font-style: normal;
  font-weight: 600;
  font-size: 30px;
  line-height: 133%;
  margin-bottom: 13px;
  
  text-decoration: underline;
  text-decoration-style: solid;
  text-underline-offset: 4px;
  text-decoration-color: hsl(248 54% 52% / 0.3);
}

.quote__text-industry:hover{
  color: hsl(248 70% 54%);
  text-decoration-color: currentColor;
}
.quote__text-industry:active{
  color: hsl(248 70% 40%);
  text-decoration-color: currentColor;
}

.quote__text-industry-icon{
  vertical-align: middle;
  margin-top: -5px;
  height: 32px;
  max-width: 36px;
  margin-left: 0.5ch;
}
```

timabbott commented on static/images/landing-page/hello/apple-touch-icon.png:
There seem to be a bunch of new icon files for Zulip logo variants that should just use the existing logo assets.

amanagr replied:
@terpimost can you review the icons? For example, `apple-touch-icon` is similar to `favicon.png` but is not the same.

terpimost replied:
I intentionally made new set of icons from the logo. Should I review and publish the reasoning? Now I can't come up with that and it will take me some time to revisit.

timabbott replied:
I think mostly what I want is for these files to have a set of descriptive names that explain what is unique about the variant; without that, it'll be hard to figure out when we should use what version. For example, here's an the existing set of logo files that we have:

```
$ ls static/images/logo/
apple-touch-icon-precomposed.png  zulip-icon-128x128.png  zulip-icon-circle.svg  zulip-org-logo.svg
README.md                         zulip-icon-512x512.png  zulip-icon-square.svg
```

It's very easy to end up with lots of duplicate and semi-duplicate versions of things that are hard to parse out which to use where without some sort of naming/description scheme. So at the very least I want all copies of the Zulip logo to be in that directory with a sensible name; adding some text to the `README.md` in that directory explaining the reasoning would be super useful if there's some subtlety to it (You can probably post a prose version and Aman can record it there) .

terpimost replied:
> I think mostly what I want is for these files to have a set of descriptive names that explain what is unique about the variant; without that, it'll be hard to figure out when we should use what version. 

> It's very easy to end up with lots of duplicate and semi-duplicate versions of things that are hard to parse out which to use where without some sort of naming/description scheme. So at the very least I want all copies of the Zulip logo to be in that directory with a sensible name; adding some text to the `README.md` in that directory explaining the reasoning would be super useful if there's some subtlety to it (You can probably post a prose version and Aman can record it there) .

@timabbott and @amanagr I don't want to fight proposed icons change or the way you store/describe things.
The icons I was proposing are not just LOGO they are the favicons of the marketing website, they are compressed in this way, sized in this way and named in this way. I don't recommend to store them in some other folder unless it is `website_favicons`

I made `manifest.webmanifest` which points to `icon-192.png` `icon-512.png` - these are the sizes recommended to be prepared for PWA (I guess apple and other companies) when you install/add app on your devices. 
There is also `apple-touch-icon` 180x180 which is used by iOS when user ads bookmark to a website to a home screen.

> A web app manifest is a JSON file that defines how a PWA should be treated as an installed application. If a website doesn't have a webmanifest file, a device can't determine the website icon and will display a default one, which can be bad for user experience.

`favicon.png` is 32x32 vs traditional 16x16, because these days high screen resolution and browsers can use that, so the icon would look in a better quality.

`favicon.ico` having 3 version of a size inside: 48, 32, 16 which should be enough to satisfy all kinds of Windows/Microsoft icon consumers:

<img width="810" alt="image" src="https://github.com/zulip/zulip/assets/1903309/65960023-d54b-41e3-b391-ab65a362dcda">

timabbott commented on web/src/portico/landing-page.js:
```diff
@@ -119,78 +172,16 @@ const apps_events = function () {
-    // Initiate the bootstrap carousel logic
-    $(".carousel").carousel({
-        interval: false,
-    });
```
After this commit, we can add a follow-up that removes all the carousel code from `web/third/bootstrap/css/bootstrap.css` and `web/third/bootstrap/js/bootstrap.js`, since this is the only place we were using that Bootstrap feature.

amanagr replied:
Yeah.

timabbott replied:
Added a couple commits to this PR doing this so that we can resolve this comment.

timabbott commented on web/styles/portico/svg_icons.css:
```diff
@@ -0,0 +1,105 @@
+        url("data:image/svg+xml,%3csvg width='140' height='32' viewBox='0 0 140 32' fill='none' xmlns='http://www.w3.org/2000/svg'%3e%3cg clip-path='url(%23clip0_780_22554)'%3e%3cpath d='M31.3809 24.2774V1.80469H34.0116V21.8992H45.7017V24.2774H31.3809Z' fill='white'/%3e%3cpath d='M54.3171 7.13232C55.4472 7.13232 56.4477 7.3351 57.3184 7.74068C58.1892 8.14625 58.9117 8.71776 59.5045 9.47361C60.097 ...
+}
+
+.client-logos__logo_layershift {
```
Can we use `web/images/landing-page/logos/` for these, rather than having PNG files embedded in our CSS files?

(I'm pretty sure it was this way in the export from the design tool for no particular reason).

timabbott replied:
Can you start a thread in #frontend about this? It seems unlikely to me that this is a way we want to do things. I can imagine configuring webpack to embed the `.png` files into a CSS file if that's what we want, but it doesn't make sense to me to take the source files (the third party's logo) and inline it into our CSS as the source format.

terpimost replied:
The only reason I did it because we want to have svg images of logos be loaded before any animation. I couldn't find svg version for those png logos, so I made them this ugly way.
If we can get svgs for those that would be great, if we want to store them as files - ok, but lets make sure browser download them before animation

timabbott replied:
OK this is not a blocker but I added a comment documenting what's weird about this file so we know what the story is.

timabbott commented on templates/corporate/hello.html:
```diff
@@ -1,562 +1,402 @@
+    <meta property="og:site_name" content="Zulip" />
+    <meta property="og:title" content="Zulip: Open-source team chat with topic-based threading" />
+    <meta property="og:description" content="Zulip is the only modern team chat app that is designed for both live and asynchronous conversations. Organized to help you collaborate productively and efficiently." />
+    <meta property="og:image" content="https://zulipchat.com/static/images/logo/zulip-icon-128x128.png" />
```
We should use zulip.com rather than zulipchat.com in all the URLs in this PR, I think.

amanagr replied:
Done.

timabbott commented on templates/corporate/hello.html:
```diff
@@ -1,562 +1,402 @@
+    <!-- font docs  https://developers.google.com/fonts/docs/css2 -->
+    <link rel="preconnect" href="https://fonts.googleapis.com"/>
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
+    <link href="https://fonts.googleapis.com/css2?family=Open+Sans:wght@300..700&family=Source+Sans+3:wght@300..700&display=block" as="style" rel="preload"/>
```
We generally have not made a habit of using external sources for fonts; we should bundle the fonts via webpack as do for everything else.

amanagr replied:
We should try to merge #24833 before this then.

amanagr replied:
Fixed but for open-sans, we are using `font-weight: 350` in a few places which [our package](https://www.npmjs.com/package/@fontsource/open-sans) doesn't support. So, I need @terpimost to confirm if we should change them to `300` or `400`. (Likely 400)

timabbott replied:
Done.

timabbott commented on templates/corporate/hello.html:
```diff
@@ -1,562 +1,402 @@
-                </p>
-                <br />
+        <div class="screen-1">
+            <h1>Organized team&nbsp;chat</h1>
```
I'm pretty sure these `&nbsp;`s should all just be spaced; just a quirk of the design tool export.

terpimost replied:
Well, it is not design tool export. I made it with a tool and think it it does a reasonable job. I don't want a situation when those words end. Up on a separate lines:
Organized team....
.... chat

This is not the same as 
Organized...
... team chat

So lets review these case instead of removing them all

timabbott replied:
Well I think the proposal was to keep ones where we think line-wrapping of the phrase is plausible.

I would still prefer to not cut up words in this way when it's not serving a functional purpose, because they make tasks like "search for where this string is in the codebase" hard.

timabbott commented on templates/corporate/hello.html:
```diff
@@ -1,562 +1,402 @@
+                    <a href="/try-zulip">
+                        <svg width="24" height="25" viewBox="0 0 24 25" fill="none" xmlns="http://www.w3.org/2000/svg">
+                            <path d="M12 14.9L13.225 12.225L15.9 11L13.225 9.775L12 7.1L10.775 9.775L8.1 11L10.775 12.225L12 14.9ZM3.5 18.5C3.1 18.5 2.75 18.35 2.45 18.05C2.15 17.75 2 17.4 2 17V5C2 4.6 2.15 4.25 2.45 3.95C2.75 3.65 3.1 3.5 3.5 3.5H20.5C20.9 3.5 21.25 3.65 21.55 3.95C21.85 4.25 22 4.6 22 5V17C22 17.4 21.85 17.75 21.55 18.05C21.25 18.35 20.9 18.5 20.5 18.5H3.5ZM1.75 21.5C1.53333 21 ...
+                        </svg>
```
I really don't like all the raw SVGs in the HTML -- it feels very hard to read. Feel free to start a #frontend thread about how we should do them, but even giving this a name and putting it in the CSS file would be way better for knowing what this is supposed to be.

And if this is a third-party icon, we need to record its licensing situation properly.

terpimost replied:
Icons were in html so we can recolor them, we can switch to image mask css property and have icons in css + saving colorization property

amanagr replied:
Done.

timabbott commented on templates/corporate/hello.html:
```diff
@@ -1,562 +1,402 @@
-                    <h2>Mobile</h2>
-                    <a href="{{ apps_page_url }}ios">
-                        <i class="fa fa-mobile-phone platform-icon" aria-hidden="true"></i>
+                    <a href="https://zulip.com/communities/">
```
Since this landing page will only be shown on zulip.com, links like this should probably all be relative.

amanagr replied:
Done.

terpimost commented on static/images/landing-page/hello/screen-2-mobile-1x.jpg:
It isn't super easy to recreate all of the files, but lets do what is needed. Should I attach files just in comments?

timabbott replied:
That's probably fine.

terpimost commented:
@amanagr please use the updated images as per Tims feedback, I also solved other bugs on them which @alya mentioned in the past

[updated images.zip](https://github.com/zulip/zulip/files/11870885/updated.images.zip)


All 3 icons are from google icons but the first one is custom made out of multiple.
In both themes they are just white, so we could place them in css as background images of 24x24 element (it could be <i>)

browse_activity icon for Try Zulip now 
```
background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' width='24' height='24' fill='none' viewBox='0 0 24 24'%3e%3cpath fill='white' d='m12 14.4 1.22-2.68 2.68-1.22-2.68-1.22L12 6.6l-1.22 2.68L8.1 10.5l2.68 1.22L12 14.4ZM3.5 18c-.4 0-.75-.15-1.05-.45-.3-.3-.45-.65-.45-1.05v-12c0-.4.15-.75.45-1.05.3-.3.65-.45 1.05-.45h17c.4 0 .75.15 1.05.45.3.3.45.65.45 1.05v12c0 .4-.15.75-.45 1.05-.3.3-.65.45-1.05.45h-17Zm-1.75 3a.73.73 0 0 1-.75-.75.73.73 0 0 1 .75-.75h20.5a.73.73 0 0 1 .75.75.73.73 0 0 1-.75.75H1.75Z'/%3e%3c/svg%3e");
```

visibility icon for See it in use
```
background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' width='24' height='24' fill='none' viewBox='0 0 24 24'%3e%3cg clip-path='url(%23a)'%3e%3cpath fill='white' d='M12 4.5C7 4.5 2.73 7.61 1 12a11.83 11.83 0 0 0 22 0c-1.73-4.39-6-7.5-11-7.5ZM12 17a5 5 0 1 1 0-10 5 5 0 0 1 0 10Zm0-8a3 3 0 1 0 0 6 3 3 0 0 0 0-6Z'/%3e%3c/g%3e%3cdefs%3e%3cclipPath id='a'%3e%3cpath fill='white' d='M0 0h24v24H0z'/%3e%3c/clipPath%3e%3c/defs%3e%3c/svg%3e");
```

add_box icon for Create organization
```
background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' width='24' height='24' fill='none' viewBox='0 0 24 24'%3e%3cpath fill='white' d='M12 17a.73.73 0 0 0 .75-.75v-3.5h3.5A.73.73 0 0 0 17 12a.73.73 0 0 0-.75-.75h-3.5v-3.5A.73.73 0 0 0 12 7a.73.73 0 0 0-.75.75v3.5h-3.5A.73.73 0 0 0 7 12a.73.73 0 0 0 .75.75h3.5v3.5A.73.73 0 0 0 12 17Zm-7.5 4c-.4 0-.75-.15-1.05-.45-.3-.3-.45-.65-.45-1.05v-15c0-.4.15-.75.45-1.05.3-.3.65-.45 1.05-.45h15c.4 0 .75.15 1.05.45.3.3.45.65.45 1.05v15c0 .4-.15.75-.45 1.05-.3.3-.65.45-1.05.45h-15Z'/%3e%3c/svg%3e");
```

amanagr commented:
Updated with your files.

amanagr commented:
@karlstolley can you please review this? It would really help!

amanagr commented:
Posted https://chat.zulip.org/#narrow/stream/43-automated-testing/topic/installing.20new.20dependency for build failure.

amanagr commented:
> The images adjacent "Efficient communication" and "Enterprise ready" appear squeezed horizontally on screens wider than 940px; this is probably due to a stale height value there (which I assume is being set because of lingering effects from Bootstrap's abomination of height: auto?)

Yeah, I asked Vlad to look into it, we want specific height for images so I am not sure what's the right fix.

amanagr commented:
> Logos in the top section are getting cut off at the left and right when the screen is below 675px.

Looks correct for me on firefox and chrome.
<img width="497" alt="image" src="https://github.com/zulip/zulip/assets/25124304/47314a59-a315-41ef-ae11-fbd731f2520b">

amanagr commented:
Oh! When I added `common.ts` to add `open-sans` font, I also added bootstrap code which maybe is causing these issues.

amanagr commented:
Everything should be fixed now.

terpimost commented:
_Sorry about the first feedback item, which is about the header, which I should catch prior_

1. Dropdown menu are not of the same color as active background of a button
Expected (2 bg colors): 
<img width="694" alt="image" src="https://github.com/zulip/zulip/assets/1903309/90f3bf45-1293-484f-b7fd-a79bf02e90eb">
Reality (3 bg colors):
<img width="687" alt="image" src="https://github.com/zulip/zulip/assets/1903309/3df3b2a7-ebe8-4fa3-8f7e-c697a4659601">
I'm not sure why we decided to make the background of the dropdown 100% opaque, it should be a bit transparent with a blur.

2. I couldn't catch they width/height bug with images. I guess it is fixed

3. In future we should replace 2022 badge with 2023. I asked @alya to give an access or a badge for it

4. Footer has headers as h3 elements, and that pulls default browsers styles:
<img width="1581" alt="image" src="https://github.com/zulip/zulip/assets/1903309/e7b80d52-056b-4d6f-894f-c54116cc8458">
`margin-block-start: 1em; margin-block-end: 1em;`

I know that we were discussing that area a lot but it doesn't feel right still. So here is the suggestion to change:
Decrease top margin on a footer:
```
#footer .footer__legal{
margin: 12px 0 5px;
...
```

Remove browser margins on h3 elements:
```
#footer .footer__section-title {
...
  margin-block-start: 0;
  margin-block-end: 0;
```

Increase the bottom margin on ul elements:
```
#footer .footer__section ul{
margin: 26px 0 28px 0;
}
```

Here is the current version (top space on a footer feels too big):
<img width="879" alt="image" src="https://github.com/zulip/zulip/assets/1903309/a39b231c-df79-443b-8b26-cb618db9a76d">

Here is the suggested correction:
<img width="869" alt="image" src="https://github.com/zulip/zulip/assets/1903309/489f3c57-0328-452c-9f2a-9da02c3c8972">

I guess we can solve header and footer issues in separate PRs... Unless some bootstrap somewhere doing something wrong... which I don't expect to be on the home page at all.

5. I noticed 1px visual bug on some dark theme mobile images:
<img width="100" alt="image" src="https://github.com/zulip/zulip/assets/1903309/00e6e28a-bfd7-46ca-95d3-bf60d936472e">

So here is the archive with updated screens 

[dark mobile updated images.zip](https://github.com/zulip/zulip/files/11991413/dark.mobile.updated.images.zip)

amanagr commented:
>  I'm not sure why we decided to make the background of the dropdown 100% opaque, it should be a bit transparent with a blur.

We decided to make it opaque until rest of the pages were migrated to the new theme. I added a commit which changes it back to semi-transparent color. Please test other pages like `/plans` to see how it looks otherwise we can decide to only switch to transparent for hello page.

> I couldn't catch they width/height bug with images. I guess it is fixed

Yes

> In future we should replace 2022 badge with 2023. I asked @alya to give an access or a badge for it

Yeah

* Updated the footer too.

> I noticed 1px visual bug on some dark theme mobile images:

Updated with new images.

karlstolley commented:
This is really clever, but probably a little too clever; I worry that users are going to interpret these as part of the interface, and not even "see" them--let alone click on them.

![Screen Shot 2023-07-10 at 11 27 11](https://github.com/zulip/zulip/assets/170719/5863cb8c-a37c-4a7d-9e4c-2f9511615324)

Either way, these buttons too are getting clipped off the screen at ~945px.

amanagr commented on web/styles/portico/hello.css:
```diff
@@ -0,0 +1,1005 @@
+body {
+    margin: 0;
+    padding: 0;
+    width: 100%;
```
This is used in some reset stylesheets as I recall. So, I didn't think much of it. @terpimost FYI

amanagr commented:
> Logos are cut off at smaller widths:

Looks like this is happening on firefox but not on chrome.

timabbott commented:
> Please test other pages like /plans to see how it looks otherwise we can decide to only switch to transparent for hello page.

I think doing transparent only for /hello is likely to be me preference.

amanagr commented:
> This is due to zoom being used. I need to look into how I can get the same behaviour here.

Looks like I will have rewrite a portion of this logic since even using `transform: scale` along with various tricks around it doesn't fix it and makes our landing page look bad on firefox.

terpimost commented:
The big idea behind the header is that it is blur background:
<img width="1049" alt="image" src="https://github.com/zulip/zulip/assets/1903309/a6618730-3500-4ad6-807b-d5fdc8c9278d">

Without the blur transparency isn't working right

terpimost commented:
> Logos are cut off at smaller widths:

I don't understand how are you making this, I tried all different widths here https://zulip-hello-page.vladkorobov.repl.co/ and it looks ok... smaller than 320 would be cut off.

amanagr commented:
Pushed an update which fixed the `zoom` bug for just the first app screenshot. Since @terpimost  said he wants to take a look too to see if there is something simpler he can do to fix this situation, I am waiting for this feedback on this.

terpimost commented:
@amanagr I think you did a great job. Thank you very much!

I noticed one little problem with not logged in user (my bug) (Sign up is cut off):
<img width="955" alt="image" src="https://github.com/zulip/zulip/assets/1903309/0d7b907e-cc0f-4b1e-9299-4f0f3699bce1">
I guess we should switch to mobile version of menu earlier instead of 960 we should do 980 or even 1024.
we have this:
```
@media (max-width: 940px){
   .top-menu{
     display: none;
   }
  .top-menu__logo{
    margin-right: unset;
    display: inline-block;
  }
  .top-menu-mobile{
    display: block;
  }
}
```
which we can replace with this:
```
@media (max-width: 980px){
   .top-menu{
     display: none;
   }
  .top-menu__logo{
    margin-right: unset;
    display: inline-block;
  }
  .top-menu-mobile{
    display: block;
  }
}
```

Also I think we should correct that to have less of a margin (from 32px to 16px):
```
.top-menu-items-group-1 {
    margin-right: 16px;
}
```


**About capterra badge.** 
Right now we are using 2022 badge and also we are loading it from capterra website: `https://capterra.s3.amazonaws.com/assets/images/gdm-badges/CA_Badge_BestValue_2022_FullColor-Positive.png`
I don't think it is great for a client, because the image size is 205 kb.
I suggest to host our own version of a badge, and use 2023 version.
Here is the file, which is 27 kb:
[capterra-badge-2023](https://github.com/zulip/zulip/assets/1903309/ce92a184-3e75-48c6-a741-8caf6050483c)

Later we will follow up with updated row of badges, so we will have more than 1...

amanagr commented:
@terpimost updated based on your last comment and fixed the `zoom` issues except client logos which seems impossible to fix using `transform: scale` or resizing container width. I think we need a different strategy to display it. Should I wrap them early into 2 rows?

<img width="957" alt="image" src="https://github.com/zulip/zulip/assets/25124304/29fe1165-1059-4f46-b3b7-aebe40ac1ca8">

amanagr commented:
@terpimost updated!

terpimost commented:
great is there anything else we should do?

amanagr commented:
Tagging for integration review from @timabbott

timabbott commented:
Something seems not quite right with the breakpoints for the navbar; one doesn't end up being able to see the organization name even when there's plenty of space.

![image](https://github.com/zulip/zulip/assets/2746074/e527a002-fdc6-404b-a6a6-c7808d042209)

timabbott commented:
* I'm still thinking we don't want the transparency on the navbar on pages other than /hello/; at least on /features it looks bad to me in a number of scroll positions where the contrast ends up poor.

* The combination of italics/bold for highlighted words doesn't look good to me; it makes reading the copy feel choppy. What do you think about doing just bold for those?

![image](https://github.com/zulip/zulip/assets/2746074/49274df6-c759-4db3-890a-8298c902d588)

* The bolded link style in the last section felt confusing to me with how we're using a bolder white for emphasis in the previous similar paragraphs. 
![image](https://github.com/zulip/zulip/assets/2746074/29287f2e-6407-46a6-bc18-19991f877948)

@alya it'd be good for you to take a look and finalize the top-of-page copy. But I think those are the only visual notes I have. 

Doing a pass on the code now.

timabbott commented on web/styles/portico/footer.css:
```diff
@@ -43,10 +43,12 @@
         border-bottom: 0;
         margin-bottom: 0;
+        margin-block-start: 0;
+        margin-block-end: 0;
```
I can't tell what these lines do, visually; probably means it's OK but I haven't reviewed this.

terpimost replied:
margin-bottom isn't working here because of default browser margin-block-start. We have to enforce 0 margin here.

timabbott commented on templates/corporate/hello.html:
```diff
@@ -1,562 +1,385 @@
+    <!--   favicon -->
+    <link rel="icon" href="{{ static('images/landing-page/hello/favicon.ico') }}" sizes="any"/> <!-- 32×32 -->
+    <link rel="apple-touch-icon" href="{{ static('images/landing-page/hello/apple-touch-icon.png') }}"/> <!-- 180×180 -->
+    {% set entrypoint = entrypoint|default("portico") %}
```
Is this a duplicate of `zerver/portico-header.html` or `templates/zerver/meta_tags.html`? IT would be nice to not duplicate those -- what's the thinking on duplicating rather than using that existing variable-based infrastructure?

timabbott commented on templates/corporate/hello.html:
```diff
@@ -1,562 +1,385 @@
+                      srcset="{{ static('images/landing-page/hello/screen-1-1x.jpg') }},
+                      {{ static('images/landing-page/hello/screen-1-2x.jpg') }} 2x,
+                      {{ static('images/landing-page/hello/screen-1-3x.jpg') }} 3x"/>
+                </picture>
```
`du -shc static/images/landing-page/hello/screen*` finds this is adding 5.7MB of images, which is quite significant compared to the total size of the codebase (and carries a real cost). Do we actually need all of these sizes? If so, it's probably bearable as a cost, but we should probably have a conversation about what version of the screenshots we want to do so that we don't have to swap them out repeatedly -- the current draft in this PR has enough differences from the current app that we might not want this precise version to go live on zulip.com in the next week.

timabbott commented on templates/corporate/hello.html:
```diff
@@ -1,562 +1,385 @@
+                                <a href="/for/communities/">Communities</a>
+                                <svg class="quote__text-industry-icon _wider" viewBox="0 0 40 32" fill="none" xmlns="http://www.w3.org/2000/svg">
+                                    <path d="M20 16.25C22.7167 16.25 25.1167 16.9 27.0667 17.75C28.8667 18.55 30 20.35 30 22.3V25H10V22.3167C10 20.35 11.1333 18.55 12.9333 17.7667C14.8833 16.9 17.2833 16.25 20 16.25ZM6.66667 16.6667C8.5 16.6667 10 15.1667 10 13.3333C10 11.5 8.5 10 6.66667 10C4.83333 10 3.33333 11.5 3.33333 13.3333C3.33333 15.1667 4.83333 16.6667 6.66667 16.6667ZM8.55 18.5C7.93333 ...
+                                </svg>
```
Should these SVGs be in the CSS file too?

terpimost replied:
It should, just size wasn't standard. @amanagr could you put it in css similar to other?

amanagr replied:
Done

timabbott commented on templates/corporate/hello.html:
```diff
@@ -1,562 +1,385 @@
+                      {{ static('images/landing-page/hello/screen-1-3x.jpg') }} 3x"/>
+                </picture>
+                <div class="cta-buttons">
+                    <a href="/try-zulip/">
```
Fixed this URL to have a trailing `/`; this saves a redirect.

amanagr replied:
Thanks

shameondev commented:
> The combination of italics/bold for highlighted words doesn't look good to me; it makes reading the copy feel choppy. What do you think about doing just bold for those?

I agree that using bold alone might look better.

terpimost commented:
> bolded link style in the last section felt confusing to me with how we're using a bolder white for emphasis in the previous similar paragraphs

It is still an emphasis and secondary is a link. There is no expectation that many people are going to click on those, but they can.

> What do you think about doing just bold for those?

Sure not problem, lets make them as "bold" in other places here (which is medium not actual bold I think)

<img width="823" alt="image" src="https://github.com/zulip/zulip/assets/1903309/f913a92f-39c2-4a64-8e99-7a418312f3b6">

terpimost commented:
> I'm still thinking we don't want the transparency on the navbar on pages other than /hello/; at least on /features it looks bad to me in a number of scroll positions where the contrast ends up poor.

Ok, here is the sugestion:
1. Clicked tab should always match background of dropdown, this situation is bad (currently on a website).
<img width="892" alt="image" src="https://github.com/zulip/zulip/assets/1903309/75cd810b-e070-4c55-b228-e92041923e94">
while this (is in PR) situation is good:
<img width="780" alt="image" src="https://github.com/zulip/zulip/assets/1903309/659aa98c-f3d5-4e21-9228-3b8b39ab65ea">


2. I don't see any case when current PR header is bad, could you please point me to such @timabbott 
<img width="1496" alt="image" src="https://github.com/zulip/zulip/assets/1903309/cc05a17f-935e-4e2f-b845-26896aa5dbf2"> has enough of a contrast.

Subjectively you might argue about level of contrast... so here is the suggestion (lets decrease transparency on hover):

```
.top-menu {
...
    background: hsla(240, 100%, 10%, 0.6);
...
    transition: background-color 0.3s ease-out;
}

.top-menu:hover {
    background: hsla(240, 100%, 10%, 0.85);
}
```

![CleanShot 2023-07-18 at 11 28 07](https://github.com/zulip/zulip/assets/1903309/08ff3979-dde7-4d9d-8a5e-7cbf7b61516f)

timabbott commented:
Decreasing transparency on hover seems like it addresses my concern there, yeah.

sbansal1999 commented on web/src/portico/landing-page.js:
```diff
@@ -14,6 +14,59 @@ export function path_parts() {
 }
 
+const hello_events = function () {
+    function getNewRand(oldRand, max) {
```
I believe the variable and function names in this code should be in snake_case instead of camelCase.

amanagr replied:
Thanks! Done

amanagr commented:
Updated with all the requested changes!

amanagr commented:
@timabbott this is ready for another review from you.

timabbott commented:
Merged the dependency commit as 724e76ee80d8e87d33ad87887d0fc47a6d9d8f63 to stop having merge conflicts when we add other dependencies.

timabbott commented:
Also merged the footer/navbar commits as the series ending with 93e6de62bc2c067298c13e773e761543d12ba5ed, after changing one of the margins from 26px to 20px from manual testing with @alya. So these changes are not isolated to just this page.

timabbott commented on web/styles/portico/navbar.css:
```diff
@@ -266,12 +274,6 @@ details summary::-webkit-details-marker {
-        max-width: 84px;
-    }
-}
-
```
Should these navbar changes be extracted to a separate commit? I hadn't noticed you had navbar changes in the giant commit, and I'm not sure what these do.

amanagr replied:
Extracted into prep commits.

timabbott commented on templates/zerver/base.html:
```diff
@@ -14,8 +14,9 @@
-        <link rel="alternate icon" href="{{ static('images/favicon.png') }}?v=4" />
+        <link id="favicon" rel="icon" href="{{ static('images/landing-page/hello/favicon.ico') }}" /> <!-- 32×32 -->
+        <link rel="alternate icon" href="{{ static('images/landing-page/hello/favicon.png') }}" />
+        <link rel="apple-touch-icon" href="{{ static('images/landing-page/hello/apple-touch-icon.png') }}"/> <!-- 180×180 -->
```
So these change are to `base.html`, which gets included everywhere -- can you do them in a separate commit? I'm also not sure we want to have these icons be under `images/landing-page/hello/` -- since they are common things for all pages.

It's really really helpful to have the things that can be reviewed as "the /hello page looks great" separated from things that impact other pages, since 95% of the risk from this sort of PR is breaking something we didn't think to test.

amanagr replied:
Done.

timabbott replied:
Thanks, this helped reveal that this change has more implications than I'd imagined.

I think in any case, anything in `base.html` should not reference `images/landing-page` -- that's common configuration for every page in Zulip, so it should be in places like `images/logo/`.

andersk replied:
This changes the primary favicon from vector to bitmap, which seems like a strict regression.

timabbott commented:
OK I did another round of review; sorry for all the delay on this. I believe the only outstanding items for this are:
* "Where your catch up on important conversations while ignoring irrelevant ones" is I think not the final subheading text; we should figure out what that should be and replace it.
* The screenshots need some updates, most significantly for the "Private messages" => "Direct messages" rename.
* The couple items above about changes to other pages in this PR that I don't understand.

Otherwise, this lgtm!

amanagr commented on web/styles/portico/navbar.css:
```diff
@@ -17,6 +17,11 @@ details summary::-webkit-details-marker {
+    transition: background-color 0.3s ease-out;
+
+    &:hover {
+        background: hsla(240deg 100% 10% / 85%);
```
<img width="1042" alt="Screenshot 2023-09-28 at 8 14 12 PM" src="https://github.com/zulip/zulip/assets/25124304/2945e2e1-ed9c-498a-805e-b1070f6963fa">

amanagr commented:
Next step is wait for @terpimost to provide updated screenshots.

timabbott commented:
Merged the first two commits (to the navbar) as the series ending with 7b252572b0bba9c5ab3830b9c5210b82d8c5d9f2, thanks for splitting those out! Having the favicon/touch-icon stuff split out is super helpful, as it made it a lot easier for me to realize those changes might not be intended.

timabbott commented:
Merged, thanks @amanagr and @terpimost for all the hard work on this!
